### ISSUE TITLE
TODO: Isolate a node in just one small container

### ISSUE THREAD

**Issue by nunojusto** [CityOfZion/neo-privatenet-docker#52]
* Description:
Install only one node in a single container and get rid of all possible stuff around.

* Objective
To have a really small docker image, that can be used after with docker compose.

**Comment by nunojusto** [comment]
-= Assigned to nunojusto =-

**Comment by nunojusto** [comment]
Any objections to use Alpine linux instead of ubuntu?

**Comment by metachris** [comment]
What do you mean with "use only one node in a single container"? This project is designed to run a consensus network of 4 nodes in parallel, inside a single container.

What's your specific goal? If you want to just run a single neo-cli node, see also https://github.com/neo-project/neo-cli/blob/master/ci/Dockerfile

**Comment by nunojusto** [comment]
Well, I was talking with @revett and the first reason behind this is to reduce the big image size of the privatenet.
So, in the end the privatenet container that we know today, can be itself a compose of 4x isolated container nodes.

**Comment by nunojusto** [comment]
But we can also stick with 4x nodes in the same container and work to get rid of the sdk and run them only in the runtime. And also transform it to Alpine linux. Maybe
In that case, the title of the issue is wrong.

**Comment by metachris** [comment]
Hm, this has been discussed and considered before, and this neo-privatenet-docker repo and the corresponding Docker image will stay with 4 nodes running in a single container. 

Not sure what you mean with "get rid of the sdk and run them only in the runtime".

Also I personally don't really feel a need to convert to Alpine, although you are free to experiment with it of course. If it significantly reduces the image size, might be interesting.

**Comment by nunojusto** [comment]
@metachris , I will do the following.
Close this issue, because 4x containers is out of scope of this project (as i see).
Work on a solution to reduce the file size significantly and then open an new issue/PR here to be evaluated.
